src/Bladeburner/Actions/TeamCasualties.ts

```diff
@@ -0,0 +1,49 @@
+import type { getRandomIntInclusive } from "../../utils/helpers/getRandomIntInclusive";
+
+export enum CasualtyFactor {
+  LOW_CASUALTIES = 0.5, // 50%
+  HIGH_CASUALTIES = 1, // 100%
+}
+
+export interface OperationTeam {
+  teamSize: number;
+  teamLost: number;
+
+  killSupportingSleeves(sleeveDeaths: number): void;
+}
+
+/**
+ * Some actions (Operations and Black Operations) use teams for success bonus
+ * and may result in casualties, reducing the player's hp, killing team members
+ * and killing sleeves (to shock them, sleeves are immortal)
+ *
+ * Caveats: TeamSize = Human Team + Supporting Sleeves set to this action
+ */
+export class TeamCasualties {
+  constructor(private severity: CasualtyFactor, private teamSizeUsed: number, private supportingSleeves: number) {}
+
+  rollOutcome(random: typeof getRandomIntInclusive, team: OperationTeam) {
+    const deaths = random(this.bestCase, this.worstCase);
+    const humans = this.teamSizeUsed - this.supportingSleeves;
+    const humanDeaths = Math.min(humans, deaths);
+    const damagedSleeves = deaths - humanDeaths;
+
+    /** Supporting Sleeves take damage when they are part of losses,
+     *   e.g. 8 sleeves + 3 team members with 4 losses -> 1 sleeve takes damage */
+    team.killSupportingSleeves(damagedSleeves);
+
+    team.teamSize -= humanDeaths;
+    team.teamLost += deaths;
```

Comment:
The current implementation is wrong. It was reported, but nobody bothers fixing it. It's best to fix it here.

Currently, it's:
```js
this.teamSize -= losses;
if (this.teamSize < this.sleeveSize) {
  const sup = Player.sleeves.filter((x) => isSleeveSupportWork(x.currentWork));
  for (let i = 0; i > this.teamSize - this.sleeveSize; i--) {
    const r = Math.floor(Math.random() * sup.length);
    sup[r].takeDamage(sup[r].hp.max);
    sup.splice(r, 1);
  }
  this.teamSize += this.sleeveSize;
}
this.teamLost += losses;
```
`this.teamSize -= losses;` is fine, but `this.teamSize += this.sleeveSize` is wrong.

Let's check this example: We have 2 team members (from recruitment), 4 supporting sleeves and we set team count to 2 when doing the operation. `this.teamSize` = 6. `this.sleeveSize` = 4. `teamSizeUsed` = 2.
If `deaths` = 2 (all team members die; 0 supporting sleeves take damage), `this.teamSize` should be 4 (0 supporting sleeves take damage; all supporting sleeves still support the main body), but:
- In the current implementation, it's 8: `this.teamSize - losses + this.sleeveSize` = 6 - 2 + 4 = 8.
- In your implementation, it's also 8: `team.teamSize - Math.min(teamSizeUsed - supportingSleeves, deaths)` = 6 - min(2 - 4, 2) = 6 - min(-2, 2) = 8.

The proper fix **may** be:
```js
// Deduct the losses
this.teamSize -= losses;
// If this happens, all team members died and there are "damaged sleeves". In this case, teamSize = sleeveSize.
if (this.teamSize < this.sleeveSize) {
  this.teamSize = this.sleeveSize;
}
team.teamLost += losses;
```
@d0sboots Please double-check this logic.

Reply:
This PR is a refactor. I'm not doing any bugfixes and I'm testing any behavior as-is, including test-covering currently known bugs.

Reply:
FWIW I agree with the stance of having this being refactor-only, and doing any bugfixes either before or after.

Reply:
Side-note, If there's an open issue on this change, I'd love to reference it in the future PR so we can keep the discussions separate.